mlrun/builder.py

```diff
@@ -51,18 +53,31 @@ def make_dockerfile(
     for build_arg_key, build_arg_value in build_args.items():
         dock += f"ARG {build_arg_key}={build_arg_value}\n"
 
-    if src_dir:
+    if source:
         dock += f"RUN mkdir -p {workdir}\n"
         dock += f"WORKDIR {workdir}\n"
-        dock += f"ADD {src_dir} {workdir}\n"
+        # 'ADD' command does not extract zip files - add extraction stage to the dockerfile
+        if source.endswith(".zip"):
+            stage1 = f"FROM {base_image} AS extractor\n"
+            stage1 += "RUN apt-get update && apt-get -y upgrade && apt install unzip\n"
+            stage1 += "RUN mkdir -p /source\n"
+            stage1 += f"COPY {source} /source\n"
+            stage1 += f"RUN cd /source && unzip {source} && rm {source}\n"
+
+            dock += f"COPY --from=extractor /source/ {workdir}\n"
+            dock = stage1 + "\n" + dock
```

Comment:
why not something like
stage1 = f"""
FROM {somewhere}
ADD {something}
RUN apt-get update ..
"""
dock = stage1 + "\n" + dock

that way, it would be a bit more easy to read